docs/gatsby-config.js

```diff
@@ -94,6 +94,15 @@ module.exports = {
           }
         }
       }
+    },
+    {
+      resolve: `gatsby-source-git`,
+      options: {
+        name: `v16`,
+        remote: `https://github.com/octokit/rest.js.git`,
+        branch: `master`, // TODO: change to version-16
+        patterns: `docs/src/pages/api/**`
+      }
```

Comment:
There is a version `15.x`, and I've pushed `16.x` for future v16 maintenance releases.

I'm okay with maintaining this config manually, but if we could make it dynamic in future, that'd be pretty cool.

Reply:
Sounds good! We could implement this similar to how it's done in [Apollo's docs theme](https://github.com/apollographql/gatsby-theme-apollo/blob/master/packages/gatsby-theme-apollo-docs/gatsby-config.js#L99-L111) to make it _more_ dynamic. It wouldn't be automatically detecting what version branches are available, but it allows us to specify versions in a way that is easier to configure.

```js
const versions = [16];

module.exports = {
  plugins: [
    ...versions.map(version => ({
      resolve: 'gatsby-source-git',
      options: {
        name: 'v' + version,
        remote: `https://github.com/octokit/rest.js.git`,
        branch: version + '.x',
        patterns: `docs/src/pages/api/**`
      }
    }))
  ]
};
```

Reply:
Can you add an entry for `beta`? Right now it would just be `16.x` and `beta`. Once version 17 is out, it is going to be `16.x` and `master`.

Reply:
We only need to add `gatsby-source-git` entries for older versions. The current version is always built from local files. So these changes will be merged into `beta` when they're ready, and then once `beta` is merged to `master`, everything will continue to work without changing anything.